### DISCUSSION TITLE
Tunneling to agate

### DISCUSSION THREAD

**Discussion by ghost** [mbrubeck/agate#111] [Q&A] [answered]
I'm using a tunnel from openport.io. First I start the server:

`bin/agate --content gemini/ --hostname openport.io --lang enUS --addr 192.168.1.101:1965 &`

Then openport.io gives me their domain:port to my IP:1965. In a remote client I use 
`gemini://openport.io:<port>`

According to their docs, that should be enough, but the server gives this message :

`WARNING dial tcp [::1]:1965: connect: connection refused`

Any ideas? It looks like an ipv6 format.

**Comment by Johann150** [comment, marked as answer]
I think your `--addr` flag is set wrong. Because you specified the devices external IP, Agate is not listening on the loopback interface (`::1` being the loopback IPv6 address). Additionally, you are not listening on IPv6 at all.

Try leaving out the `--addr` flag, then Agate should be listening in dual stack mode on all interfaces on the default port.

**Reply by Johann150** [reply]
I might also add, that your `--lang` flag is also wrong, as listed by the help text of the program (using `--help`) it should be an RFC 4646 language code to be compatible with Gemini, you are missing the dash for `en-US`.

**Comment by ghost** [comment]
Thanks again Johann, that solves it!
